File: hw/maisincho/u3/HW17MazeSolid/MazePython/model/door.py
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from model.room import Room  

class Door:
    def __init__(self, room1: 'Room', room2: 'Room'):
        self._room1 = room1
        self._room2 = room2

    @property
    def room1(self) -> 'Room':
        return self._room1

    @property
    def room2(self) -> 'Room':
        return self._room2

File: hw/suarez/U3/HW17MazeGeneratorASCII/MazeGenerator/src/ec/edu/espe/model/Door.java
package ec.edu.espe.model;

public class Door extends Wall {
    private Room room1;
    private Room room2;

    public Door(Room room1, Room room2) {
        super();
        this.room1 = room1;
        this.room2 = room2;
    }

    public Room getOtherRoom(Room current) {
        return current == room1 ? room2 : room1;
    }
}
